src/java.desktop/share/native/liblcms/LCMS.c

```diff
@@ -756,11 +756,14 @@ static cmsHPROFILE _writeCookedTag(const cmsHPROFILE pfTarget,
         const void* pTag = cmsReadTag(pfSanity, sig);
         if (pTag == NULL) {
             // the tag can not be cooked
+            cmsCloseProfile(pfSanity);
             cmsCloseProfile(p);
-            p = NULL;
+            return NULL;
         }
-        cmsCloseProfile(pfSanity);
-        pfSanity = NULL;
+        // The profile we used for sanity checking needs to be returned
+        // since the one we updated is raw - not cooked.
+        cmsCloseProfile(p);
```

Comment:
Looks like now we close this profile in all branches, so can we do that just above the "if (pfSanity == NULL)"?

Reply:
Ok

Reply:
Ok